### ISSUE TITLE
Is it also possible to bind the images received from the backend in your module?

### ISSUE THREAD

**Issue by Apps-23** [aberezkin/ng2-image-upload#12]
Suppose I receive 10 images , out of those 10 I replace two specific images with new images and send back the server call

**Comment by aberezkin** [comment]
Each image is sent to the server individually.

**Comment by aberezkin** [comment]
I don't think that there's a way to replace uploaded images with the ones that you got from the backend. Can you tell what is the purpose of such feature? Maybe if this is a popular use-case I can create an interface for this.

**Comment by Apps-23** [comment]
The scenario is:
I use your module to upload a set of images in "add a new user" functionality.After that the array of images is sent to the server on the final call (while on each (onFileUploadFinish) event I push them in the array)
Now, when I open the same form in "Edit the Existing User" I want to prepopulate your component with the array of images I uploaded and now received from the server.

**Comment by aberezkin** [comment]
Yes, this scenario is reasonable. I'm considering to add this functionality but this will definitely make the interface more sophisticated. I'm trying to keep the balance between functionality and ease of use.

What do you think, how interface for this scenario should look?

**Comment by Apps-23** [comment]
Hey aberezkin!
I am naive with this technology though I believe we can use an ngFor to iterate over the urls received in a seperate section.
One more query how can I differentiate between two uploads on the same the same page?

**Comment by FlyingKatinka** [comment]
I have the same issue. I also want to display the already uploaded image when I edit the current entry. Would be great if you would implement a solution. Thanks...

**Comment by miklblitz** [comment]
if resove binding uploaded images, it is will be very cool, i have same problem.

**Comment by miklblitz** [comment]
yes i've done it. Here other varian with uploaded photos https://github.com/miklblitz/angular2-image-upload_uploaded

**Comment by akolybelnikov** [comment]
how to use the new variant with this functionality? the npm module installs the old version and upon setting up the files in the node_modules, the files look differently and the structure of the module does not resemble the one in the git repo. I have been digging trying to add the functionality but so far no success.

**Comment by fergalmoran** [comment]
This is an essential feature.

**Comment by UncleDave** [comment]
If I'm not mistaken you can do this now.
Have not tested this so if I'm incorrect please reopen.

Assuming your server responds with a file - using the `onFileUploadFinish` output it should be possible to do the following:

````
onFileUploadFinish(fileholder: FileHolder) {
  fileholder.file = fileholder.response;
  fileholder.src = URL.createObjectURL(fileholder.response);
}
````
Or if your server responds with the file and the URL:

````
onFileUploadFinish(fileholder: FileHolder) {
  const resp = fileholder.response.json();

  fileholder.file = resp.file;
  fileholder.src = resp.url;
}
````

**Comment by aberezkin** [comment]
@UncleDave no, I don't think that this is what @Apps-23 means. Suppose that you have a blog post, and it has some images, but you want to edit it and upload some other images. To do that you need to get ulrs of these images from the server put them to the `ImageUploadComponent` and after that delete some old images and upload some new. To do that you need an interface (input for example) to bind these images in the component.

**Comment by UncleDave** [comment]
That is true - if the feature is to get some images on open.

**Comment by a-kolybelnikov** [comment]
@Apps-23 "(while on each (onFileUploadFinish) event I push them in the array)" - what's the way to do that? I get separate events with one event.file each. onFileUpload Finish gets called every time for each event.
Thanks!

**Comment by sabrio** [comment]
+1, is somebody working on this?

**Comment by UncleDave** [comment]
Closed by #130, thanks to @sabrio
